source/ui/syntaxhighlighter.cpp

```diff
@@ -1,38 +1,98 @@
-* Copyright (C) 2016 Chair of Computer Architecture
-* at Technical University of Munich
-*
-* This program is free software: you can redistribute it and/or modify
-* it under the terms of the GNU General Public License as published by
-* the Free Software Foundation, either version 3 of the License, or
-* (at your option) any later version.
-*
-* This program is distributed in the hope that it will be useful,
-* but WITHOUT ANY WARRANTY; without even the implied warranty of
-* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-* GNU General Public License for more details.
-*
-* You should have received a copy of the GNU General Public License
-* along with this program. If not, see <http://www.gnu.org/licenses/>.
-*/
-
-#include "ui/syntaxhighlighter.hpp"
-#include <iostream>
-
-SyntaxHighlighter::SyntaxHighlighter(std::vector<KeywordRule> &&keywords,
-                                     QTextDocument *document)
-: QSyntaxHighlighter(document), _keywords(std::move(keywords)) {
-}
-
-void SyntaxHighlighter::highlightBlock(const QString &text) {
-  for (uint i = 0; i < _keywords.size(); i++) {
-    const KeywordRule &rule = _keywords.at(i);
-    QRegularExpressionMatchIterator it = rule.rulePattern.globalMatch(text);
-    while (it.hasNext()) {
-      QRegularExpressionMatch match = it.next();
-      setFormat(
-          match.capturedStart(), match.capturedLength(), rule.ruleTextFormat);
-    }
-  }
-}
+* C++ Assembler Interpreter
+* Copyright (C) 2016 Chair of Computer Architecture
+* at Technical University of Munich
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "ui/syntaxhighlighter.hpp"
+#include <QQuickTextDocument>
+#include <QString>
+#include <QTextCharFormat>
+#include <iostream>
+#include "core/parser-interface.hpp"
+
+SyntaxHighlighter::SyntaxHighlighter(ParserInterface parserInterface,
+                                     QTextDocument *document)
+: QSyntaxHighlighter(document) {
+  // Add all instruction keywords to the syntax highlighter
+  QTextCharFormat instructionFormat;
+  instructionFormat.setForeground(Qt::darkBlue);
+  instructionFormat.setFontWeight(QFont::Bold);
+  _addKeywords(SyntaxInformation::Token::Instruction,
+               instructionFormat,
+               QRegularExpression::CaseInsensitiveOption,
+               parserInterface);
+
+  // Add the immediate regex to the syntax highlighter

```

Comment:
Personally, I would put these sections into functions. Why? Because I think comments should be largely unnecessary in readable code. For example, you could write this like so:
```
_addImmediateRegexToSyntaxHighlighter(parserInterface);
```

That would make this constructor much smaller, and you don't need a comment anymore! Because the code is the comment.